bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/InvalidationTree.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.mps.model.sync.bulk
+
+import gnu.trove.map.TLongObjectMap
+import gnu.trove.map.hash.TLongObjectHashMap
+import org.modelix.model.api.INode
+import org.modelix.model.api.ITree
+import org.modelix.model.api.PNodeAdapter
+import org.modelix.model.api.ancestorsAndSelf
+import org.modelix.model.sync.bulk.ModelSynchronizer
+
+/**
+ * The purpose of this data structure is to store which nodes changes and need to be synchronized,
```

Comment:
```suggestion
 * The purpose of this data structure is to store which nodes change and need to be synchronized,
```

? Or what is the correct grammar / meaning of this sentence?